File: 53-maximum-subarray/53-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        
        int currBest=nums[0],overAllBest=nums[0];
        
        for(int i=1;i<nums.size();i++){
        
            if(nums[i]>currBest+nums[i])
                currBest=nums[i];
            else
                currBest+=nums[i];
            
            overAllBest=max(overAllBest,currBest);
        }
        
        return overAllBest;
    }
};

File: 53-maximum-subarray/53-maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let currBest=nums[0],overallBest=nums[0]
    for(let i=1;i<nums.length;i++){
        if(nums[i]>nums[i]+currBest)
            currBest=nums[i];
        else
            currBest+=nums[i];
        overallBest=Math.max(overallBest,currBest);
    }
    return overallBest;
};